src/cpu_shader/flatten.rs

```diff
@@ -151,12 +311,176 @@ fn flatten_cubic(cubic: Cubic, line_ix: &mut usize, lines: &mut [LineSoup]) {
     }
 }
 
+fn flatten_arc(
+    path_ix: u32,
+    begin: Vec2,
+    end: Vec2,
+    center: Vec2,
+    angle: f32,
+    transform: &Transform,
+    line_ix: &mut usize,
+    lines: &mut [LineSoup],
+    bbox: &mut IntBbox,
+) {
+    let mut p0 = transform.apply(begin);
+    let mut r = begin - center;
+    let tol: f32 = 0.5;
+    let radius = tol.max((p0 - transform.apply(center)).length());
+    let x = 1. - tol / radius;
+    let theta = (2. * x * x - 1.).clamp(-1., 1.).acos();
+    const MAX_LINES: u32 = 1000;
+    let n_lines = if theta <= ROBUST_EPSILON {
+        MAX_LINES
+    } else {
+        MAX_LINES.min((std::f32::consts::TAU / theta).ceil() as u32)
+    };
+
+    let th = angle / (n_lines as f32);
```

Comment:
Why does the number of lines depend only on the radius, rather than including the angle being swept over?

I would naïvely expect `th` to be the same for all circles with the same radius, as that should make equal length lines.
It feels quite likely that I'm missing something obvious here

Reply:
I also have this concern, and wonder if it might have something to do with the high observed memory usage.

Reply:
Ah yes, there is a bug here and thank you for taking a close look and catching that. `n_lines` is the number for a full circle. This is directly transcribed from the GPU flatten code where I was getting the count for a full circle and fitting that into any sweep angle. I also noticed that the derivation of `theta` is also not quite right. Given a radius $r$ and the length $l$ of the line segment from the center of the circle to the middle of the chord (i.e. a flattened line segment) given by $l = r - tol$, the derivation should look like:

$$ tol = r - r \times cos(\frac{\theta}{2}) $$

$$ \theta = 2 \times arccos(1 - \frac{tol}{r}) $$

And `n_lines = (angle / theta).ceil()`. We still need to guard against a division by zero / over-subdividing. I went ahead and picked an arbitrary minimum angle limit of `0.0001` radians but this could probably get fine tuned.